connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -956,7 +956,7 @@ class Dataset[T] private[sql] (
     val generator = SparkUserDefinedFunction(
       UDFAdaptors.iterableOnceToSeq(f),
       UnboundRowEncoder :: Nil,
-      ScalaReflection.encoderFor[Seq[A]])
+      ScalaReflection.encoderForWithRowEncoderSupport[Seq[A]])
```

Comment:
Please keep in mind that for encoders that are used to convert data from the external representation to the internal representation we need an encoder with a valid schema. An UnboundRowEncoder does not have a valid/meaningful schema. It should not be used in these cases.

Reply:
Thanks for the comment. I evaluated all call sites, it turns out none of them need to be with Row Encoder support. I only left a change in [[ExpressionEncoder]].